aws-https/lib/aws-https-with-aws.test.ts

```diff
@@ -10,45 +10,128 @@ describe('AwsHttps with AWS', () => {
+        });
+    });
 
-        // WHEN
-        const reply = await sut.request({
+    describe('with process AWS credentials', () => {
+        const body = JSON.stringify({cursor: 3, map: "Irvine"});
+        const requestOptions = {
             protocol: 'https:',
             method: 'POST',
             hostname: serverHostname,
             path: '/test',
             body: body,
             awsSign: true,
-            headers: { 'Content-Type': 'application/json' }
+            headers: {'Content-Type': 'application/json'}
+        };
+        const response = {success: true};
+        let scope;
+
+        beforeEach(() => {
+            scope = nock(serverUrl, {
+                reqheaders: {
+                    "X-Amz-Date": /20......T......Z/,
+                    "Authorization": /AWS4-HMAC-SHA256 Credential=deadbeaf.*, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=.*/
+                }
+            }).post('/test', body)
+              .reply(200, response);
         });
 
-        // THEN
-        expect(reply).toEqual(response);
-        expect(scope.isDone()).toBeTruthy();
-        expect(AwsHttps['credentialsPromise']).toBeTruthy();
+        test('does first signed POST', async () => {
+            // // GIVEN
+            const sut = new AwsHttps(true);
+            expect(AwsHttps['credentialsInitializedPromise']).toBeUndefined();
+            expect(sut['awsCredentials']).toBeUndefined();
+
+            // WHEN
+            const reply = await sut.request(requestOptions);
 
-        // TEST refresh credentials
-        const sutReusingCreds = new AwsHttps();
-        expect(AwsHttps['credentialsPromise']).toBeDefined();
-        const sutRefreshingCreds = new AwsHttps(false, true);
-        expect(AwsHttps['credentialsPromise']).toBeUndefined();
+            // THEN
+            expect(reply).toEqual(response);
+            expect(scope.isDone()).toBeTruthy();
+            expect(AwsHttps['credentialsInitializedPromise']).toBeTruthy();
+            expect(sut['awsCredentials']).toBeDefined();
+        });
+
+        test('does another signed POST; reuse creds', async () => {
+            // // GIVEN
+            const sut = new AwsHttps(true);
+            expect(AwsHttps['credentialsInitializedPromise']).toBeDefined();
+            expect(sut['awsCredentials']).toBeUndefined();
+
+            // WHEN
+            const reply = await sut.request(requestOptions);
+
+            // THEN
+            expect(reply).toEqual(response);
+            expect(scope.isDone()).toBeTruthy();
+            expect(AwsHttps['credentialsInitializedPromise']).toBeTruthy();
+            expect(sut['awsCredentials']).toBeDefined();
+        });
+
+        test('resets credentials', async () => {
+            const sut = new AwsHttps(false, true);
+            expect(AwsHttps['credentialsInitializedPromise']).toBeUndefined();
+            expect(sut['awsCredentials']).toBeUndefined();
```

Comment:
Given, when, and then were omitted here.  Perhaps it was intentional because the test is simple.